### PR TITLE
Refactors Kube RBAC

### PR DESCRIPTION
**Pull request by danehans**
Fixes: #291 
Fixes: #289

This code failed e2e testing with:
```
2022-08-30T18:29:03.209Z	ERROR	runner/runner.go:53	failed to create new infra	{"runner": "infrastructure", "error": "failed to get serviceaccount envoy-gateway-system/eg: serviceaccounts \"eg\" is forbidden: User \"system:serviceaccount:envoy-gateway-system:envoy-gateway\" cannot get resource \"serviceaccounts\" in API group \"\" in the namespace \"envoy-gateway-system\""}
```

Signed-off-by: danehans <daneyonhansen@gmail.com>

### CODE REVIEW

File: internal/infrastructure/kubernetes/config/rbac/role.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  namespace: system
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - serviceaccounts
+      - services
+    verbs:
+      - create
+      - get
```
Comment (skriss) [inline internal/infrastructure/kubernetes/config/rbac/role.yaml:17]: perhaps need get/list/watch here? those are usually all needed for the informers to work correctly.

Comment (danehans) [reply internal/infrastructure/kubernetes/config/rbac/role.yaml:17]: @skriss that did the trick. Thank you!

**Comment by danehans** [comment]
@skriss I create a separate Infra Manager Role, attach the Role to the Envoy Gateway ServiceAccount using a RoleBinding, but I still get the same error:
```
$ kubectl get role/infra-manager -n envoy-gateway-system -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"Role","metadata":{"annotations":{},"name":"infra-manager","namespace":"envoy-gateway-system"},"rules":[{"apiGroups":[""],"resources":["serviceaccounts","services"],"verbs":["create","get","list","watch"]},{"apiGroups":["apps"],"resources":["deployments"],"verbs":["create","get","list","watch"]}]}
  creationTimestamp: "2022-09-01T18:13:42Z"
  name: infra-manager
  namespace: envoy-gateway-system
  resourceVersion: "76599"
  uid: 4d3197e5-0c3b-4038-811f-0072a03ebc8c
rules:
- apiGroups:
  - ""
  resources:
  - serviceaccounts
  - services
  verbs:
  - create
  - get
  - list
  - watch
- apiGroups:
  - apps
  resources:
  - deployments
  verbs:
  - create
  - get
  - list
  - watch

$ kubectl get rolebinding/infra-manager -n envoy-gateway-system -o yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"rbac.authorization.k8s.io/v1","kind":"RoleBinding","metadata":{"annotations":{},"name":"infra-manager","namespace":"envoy-gateway-system"},"roleRef":{"apiGroup":"rbac.authorization.k8s.io","kind":"Role","name":"infra-manager"},"subjects":[{"kind":"ServiceAccount","name":"envoy-gateway","namespace":"system"}]}
  creationTimestamp: "2022-09-01T18:13:42Z"
  name: infra-manager
  namespace: envoy-gateway-system
  resourceVersion: "76600"
  uid: 91431b52-3603-498a-bb06-97ce34a568d9
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: infra-manager
subjects:
- kind: ServiceAccount
  name: envoy-gateway
  namespace: system

$ kubectl get deploy/envoy-gateway -n envoy-gateway-system -o yaml | grep serviceAccountName:
      serviceAccountName: envoy-gateway

$ kubectl logs deploy/envoy-gateway -n envoy-gateway-system -f
...
2022-09-01T18:17:52.706Z	ERROR	runner/runner.go:53	failed to create new infra	{"runner": "infrastructure", "error": "failed to get serviceaccount envoy-gateway-system/eg: serviceaccounts \"eg\" is forbidden: User \"system:serviceaccount:envoy-gateway-system:envoy-gateway\" cannot get resource \"serviceaccounts\" in API group \"\" in the namespace \"envoy-gateway-system\""}
```

**Comment by skriss** [comment]
Looks like the namespace in the `subjects` is not correct (should be `envoy-gateway-system`, not just `system`:

```yaml
subjects:
- kind: ServiceAccount
  name: envoy-gateway
  namespace: system
```

**Comment by danehans** [comment]
@skriss I replicated the Kube provider kustomization setup for the Infra Manager and it still failed to append the `envoy-gateway-system` namespace to the RoleBinding `subjects[].namespace`. I have removed the namespace appending from the infra kustomization setup.

**Review by skriss — APPROVED** [review]
LGTM

**Pull request merged**